src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/GenerateMethodCodeActionProvider.cs

```diff
@@ -174,15 +174,35 @@ private static bool TryGetEventNameAndMethodName(
             {
                 if (attribute.Name == markupTagHelperDirectiveAttribute.TagHelperAttributeInfo.Name)
                 {
-                    if (!attribute.IsEventCallbackProperty())
+                    if (attribute.IsEventCallbackProperty())
                     {
-                        return false;
+                        // TypeName is something like "EventCallback<System.String>", so we need to parse out the parameter type.
+                        if (ComponentAttributeIntermediateNode.TryGetEventCallbackArgument(attribute.TypeName.AsMemory(), out var argument))
+                        {
+                            eventParameterType = argument.ToString();
+                        }
                     }
-
-                    // TypeName is something like "EventCallback<System.String>", so we need to parse out the parameter type.
-                    if (ComponentAttributeIntermediateNode.TryGetEventCallbackArgument(attribute.TypeName.AsMemory(), out var argument))
+                    else if (attribute.IsDelegateProperty())
                     {
-                        eventParameterType = argument.ToString();
+                        if (attribute.IsGenericTypedProperty())
```

Comment:
Do we need to have an extra `out` parameter here to prevent "Generate Async Event Handler" showing up, or does the runtime happily call `Action<T>` delegates in an async context?

Reply:
This is a good point. Async shouldn't be added for the cases I've seen. I could imagine someone allows an override of `Action<Task<T>>` but I haven't seen it in my usage.